app/javascript/dashboard/store/modules/conversations/actions.js

```diff
@@ -86,6 +86,62 @@ const actions = {
     }
   },
 
+  syncActiveConversationMessages: async (
+    { commit, state, dispatch },
+    { conversationId }
+  ) => {
+    const { allConversations, syncConversationsMessages } = state;
+    const lastMessageId = syncConversationsMessages[conversationId];
+    const selectedChat = allConversations.find(
+      conversation => conversation.id === conversationId
+    );
+    if (!selectedChat) return;
+    try {
+      const { messages } = selectedChat;
+      const {
+        data: { meta, payload },
+      } = await MessageApi.getPreviousMessages({
+        conversationId,
+        after: lastMessageId,
+      });
+      commit(`conversationMetadata/${types.SET_CONVERSATION_METADATA}`, {
+        id: conversationId,
+        data: meta,
+      });
+      const missingMessages = payload.filter(
+        message => !messages.find(item => item.id === message.id)
+      );
+      commit(types.SET_MISSING_MESSAGES, {
+        id: conversationId,
+        data: missingMessages,
+      });
+      commit(types.CLEAR_LAST_MESSAGE_ID_FROM_SYNC_CONVERSATION, {
+        conversationId,
+      });
```

Comment:
We can re-use the same mutation

```suggestion
      commit(types.SET_LAST_MESSAGE_ID_IN_SYNC_CONVERSATION, {
        conversationId,
        messageId: null
      });
```

Reply:
Sounds good.